run_xnat2bids.py

```diff
@@ -175,84 +228,122 @@ async def main():
 
-        # Remove the password parameter from the x2b_param_list
-        x2b_param_list_without_password = [param for param in x2b_param_list if not param.startswith('--pass')]
+            # Remove the password parameter from the x2b_param_list
+            x2b_param_list_without_password = [param for param in x2b_param_list if not param.startswith('--pass')]
 
-        logging.debug({
-        "message": "Argument List",
-        "session": session,
-         "slurm_param_list": slurm_param_list,
-        "x2b_param_list": x2b_param_list_without_password,
+            logging.debug({
+            "message": "Argument List",
+            "session": session,
+                "slurm_param_list": slurm_param_list,
+            "x2b_param_list": x2b_param_list_without_password,
+
+            })
+
+    # Run Heudiconv for any user defined sessions
+    if "dcm2bids-args" in arg_dict:
+        dcm2bids_bindings = []
+
+        # Fetch dcm2bids parameters
+        d2b_param_list = compile_dcm2bids_list(arg_dict, dcm2bids_bindings)
+
+        heudiconv_slurm_params = list(slurm_param_list)
+
+        # Define output for logs
+        if not ('output' in arg_dict['slurm-args']):
+            output = f"/gpfs/scratch/{user}/logs/%x-%J.txt"
+            arg = f"--output {output}"
+            heudiconv_slurm_params.append(arg)
+
+        add_job_name(heudiconv_slurm_params, "xnat-heudiconv")
+
+        # Add BIDS_ROOT to bindings for heudiconv
+        if not ("bids_root" in arg_dict["dcm2bids-args"]):
+            d2b_param_list.insert(2, bids_root)
+            dcm2bids_bindings.append(bids_root)
+
+        # Add heudiconv command arguments to list for execution
+        argument_lists.append(("xnat-heudiconv", d2b_param_list, heudiconv_slurm_params, dcm2bids_bindings))
 
-        })
 
     # Loop over argument lists for provided sessions.
     tasks = []
+    needs_dependency = False
     for args in argument_lists:
-        # Compile bindings into formated string
-        bindings = ' '.join(f"-B {path}" for path in args[2])
-
         # Compilie slurm and xnat2bids args 
-        xnat2bids_options = ' '.join(args[0])
-        slurm_options = ' '.join(args[1])
+        xnat_tools_cmd = args[0]
+        xnat2bids_options = ' '.join(args[1])
+        slurm_options = ' '.join(args[2])
+
+        # Compile bindings into formated string
+        bindings = ' '.join(f"-B {path}" for path in args[3])
 
         # Build shell script for sbatch
         sbatch_script = f"\"$(cat << EOF #!/bin/sh\n \
             apptainer exec --no-home {bindings} {simg} \
-            xnat2bids {xnat2bids_options}\nEOF\n)\""
+            {xnat_tools_cmd} {xnat2bids_options}\nEOF\n)\""
 
         # Process command string for SRUN
         sbatch_cmd = shlex.split(f"sbatch -Q {slurm_options} \
             --wrap {sbatch_script}")    
 
         # Set logging level per session verbosity. 
-        set_logging_level(args[0])
+        set_logging_level(args[1])
```

Comment:
maybe pull out `args[1]` into a variable `logging_level = args[1]` since it's used more than once